### PR TITLE
feat(argus): user defined secret changes

### PR DESCRIPTION
**Pull request by sagarmamdapure**

### CODE REVIEW

File: charts/argus/templates/_helpers.tpl [hunk 1]
```diff
@@ -130,4 +130,67 @@ Collector Pod security context
...
\ No newline at end of file
+{{- end }}
+
+{{/*
+LM Credentials and Proxy Details
+*/}}
+
+{{- define "lm-credentials-and-proxy-details" -}}
+{{- $secretObj := (lookup "v1" "Secret" .Release.Namespace .Values.global.userDefinedSecret) | default dict }}
+{{- $secretData := (get $secretObj "data") | default dict }}
```
Comment (vkumbhar94) [inline charts/argus/templates/_helpers.tpl:141]: where are we using this retrieved object?
Good to add some comments here mentioning the code flow.

File: charts/argus/templates/_helpers.tpl [hunk 2]
```diff
@@ -130,4 +130,67 @@ Collector Pod security context
...
+      key: proxyPass
+      {{- end }}
+{{- end }}
+{{- end }}
+
+
+{{/*
+Return secret name to be used based on the userDefinedSecret.
+*/}}
+{{- define "argus-secret-name" -}}
```
Comment (vkumbhar94) [inline charts/argus/templates/_helpers.tpl:190]: Can we move this to `lm-util` by renaming it to something like `lmutil.secret-name`?

Template refers the global section, nothing specific so good to have this there, so one secret to configure all child charts (looking out from usability perspective)

**Review by vkumbhar94 — CHANGES_REQUESTED** [review]

File: charts/argus/templates/_helpers.tpl (hunk 1 above)
Comment (sagarmamdapure) [reply charts/argus/templates/_helpers.tpl:144]: Noted

File: charts/argus/templates/_helpers.tpl (hunk 2 above)
Comment (sagarmamdapure) [reply charts/argus/templates/_helpers.tpl]: Make sense. Noted

**Review by shubhashri12 — APPROVED** [review]

**Pull request merged**